### ISSUE TITLE
Feature Request - LDAP JSON Export into BloodHound Format

### ISSUE THREAD

**Issue by ad0nis** [SecureAuthCorp/impacket#706]
Noticed that impacket relay to LDAP was generating information about the domain in multiple formats including JSON - got my hopes up that this would import into BloodHound, and was disappointed to find it was not compatible. Would be super handy if an initial relay to LDAP could immediately be imported into BloodHound.

**Comment by asolino** [comment]
CCing @dirkjanm :)

**Comment by martingalloar** [comment]
This is an interesting one! Would having an LDAP socks proxy help running SharpHound on a relayed connection? Or maybe is worth some thought making @dirkjanm's BloodHound.py integrate with `ntlmrelayx` somehow?

**Comment by dirkjanm** [comment]
There is one for the old (CSV) format but not yet for the 2.0 format. This request is tracked here: https://github.com/dirkjanm/ldapdomaindump/issues/14 but as very few people seem to use it I'm not sure this is something I will spend time at developing further.

@martingalloar that would be possible but for the LDAP connection BloodHound.py requires at least 2 sockets to LDAP and 1 to the GC as well in multi-domain environments.

The easiest way to go for this right now is with the feature to add a machine account via the relayed connection, which you can use to run bloodhound or other gathering tools.

**Comment by sho-luv** [comment]
I think it's only not popular because not a lot of people know ntlmrelayx.py dumps domain information when you use LDAP. I was recently in an environment where no user could add a computer account however I was able to dump the domain information with ntlmrelayx.py The passwords were too complex to crack. 

I wasn't able to run bloodhound because I didn't have valid creds. I did however have a ton of domain information thanks to ntlmrelayx.py dumping it for me. I then had the thought that lead me here to see if it would be possible to just provide an option for bloodhound format etc. I was planning on working on it for myself. Reading @dirkjanm's notes I'm not sure it's feezible. If it is I don't mind spending the time to work it out.

**Comment by choket** [comment]
I agree, this would be very useful in situations where you can relay NTLMv2 hashes but don't have creds.

**Comment by ad0nis** [comment]
So if I am reading these docs right, it looks like the tools to do this were eventually built... 

https://github.com/dirkjanm/ldapdomaindump

Any chance they can be merged back into impacket now, so that a relayed ldap connection can output bloodhound-compatible files automatically?

**Comment by gjhami** [comment]
I was considering helping with this feature, but I believe this should just be closed instead.

I think the solution is to create a SOCKS proxy to LDAP(S) using NTLMRelayx and then allow users to run a bloodhound ingestor of their choice (such as [bloodhound.py](https://github.com/dirkjanm/BloodHound.py)) using the proxy. This seems preferable to incorporating bloodhound ingestor functionality directly into impacket.
